common/src/storage_queue.rs

```diff
@@ -20,22 +20,28 @@ use codec::{Decode, Encode};
 use primitive_types::H256;
 use scale_info::TypeInfo;
 use sp_std::borrow::Cow;
+use sp_std::marker::PhantomData;
 use sp_std::prelude::*;
 
+fn key_with_prefix(prefix: &[u8], key: &[u8]) -> Vec<u8> {
+    [prefix, key].concat()
+}
+
 #[derive(Debug, Clone, Encode, Decode, TypeInfo)]
 struct Node<T: Encode + Decode> {
     value: T,
     next: Option<H256>,
 }
 
 #[derive(Debug, Clone, Encode, Decode, TypeInfo)]
-pub struct StorageQueue {
+pub struct StorageQueue<T: Decode + Encode> {
```

Comment:
Just for future beautifying - I found it more comfortable to write `T: Codec`, instead of `T: Decode + Encode`

(Provided by `parity_scale_codec` -> `pub trait Codec: Encode + Decode {}`)